pkg/common/security/usergroup_test.go

```diff
@@ -93,64 +140,80 @@ func TestBrokenUserGroup(t *testing.T) {
 	if ug.Groups[0] != "100" {
 		t.Errorf("User 'testuser2' primary group resolved while it should not: %v", ug)
 	}
+	testCache.lock.Unlock()
 
 	ug, err = testCache.GetUserGroup("testuser3")
 	if err != nil {
 		t.Error("Lookup should not have failed: testuser3")
 	}
-	if len(testCache.ugs) != 2 {
-		t.Errorf("Cache not updated should have 2 entries %d", len(testCache.ugs))
-	}
-	// check returned info: 4 groups, primary group is duplicate
-	if len(ug.Groups) != 4 {
-		t.Errorf("User 'testuser3' not resolved correctly: duplicate primary group not filtered %v", ug)
-	}
+
+	assert.Equal(t, 2, testCache.getUGsize(), "Cache not updated should have 2 entries %d", len(testCache.ugs))
+	assert.Equal(t, 4, testCache.getUGGroupSize("testuser3"), "User 'testuser3' not resolved correctly: duplicate primary group not filtered %v", ug)
+
+	ug, err = testCache.GetUserGroup("unknown")
+	assert.ErrorContains(t, err, "lookup failed for user: unknown")
+
+	ug, err = testCache.GetUserGroup("testuser4")
+	assert.NilError(t, err)
+
+	ug, err = testCache.GetUserGroup("testuser5")
+	assert.ErrorContains(t, err, "lookup failed for user: testuser5")
+
+	ug, err = testCache.GetUserGroup("invalid-gid-user")
+	testCache.lock.Lock()
+	assert.ErrorContains(t, err, "lookup failed for user: invalid-gid-user")
+	exceptedGroup := []string{"1_001"}
+	assert.Assert(t, reflect.DeepEqual(ug.Groups, exceptedGroup), fmt.Errorf("group should be: %v, but got: %v", exceptedGroup, ug.Groups))
+	testCache.lock.Unlock()
```

Comment:
This lock/unlock pair can be removed. `GetUserGroup()` returns a copy and internally uses a mutex to access the map.